### ISSUE TITLE
Squirrel macOS: can easily get into a corrupt warning state

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#28539]
Steps to Reproduce:

1. download a release that will get an update
2. wait for the update notification to appear
3. quit (do NOT apply the update, just quit)
4. start up again after 1-2 seconds
5. see a warning about VS Code installation being corrupt

Here is what happens:
* Squirrel downloads the update and shows the notification
* User quits
* Squirrel starts to apply the update
* User starts
* Squirrel has not finished applying the update, you are still on the old workbench
* after 10s our integration check kicks in
* by now Squirrel has finished the update
* we show an integrity failure warning because the files have actually changed while we were running

Bottom line: this does NOT reproduce when the user decides to apply the update where we then call `quitAndInstall()` which seems to do everything nicely in one call. What we actually need is a `quitAndInstall` method that does not restart which we should always call when an update is available to prevent this (only really quit when the update is installed).

One thing I verified is that Squirrel is NOT replacing files while Code is still running, only after quit.

/cc @alexandrudima maybe we need to tweak this warning message to ask the user for a restart first to see if that fixes it

**Comment by alexandrudima** [comment]
I'm open for changing the message. Please directly push a change to the warning message in https://github.com/Microsoft/vscode/blob/3d3167ace71701c63da04f7e2905f83401af10fa/src/vs/platform/integrity/node/integrityServiceImpl.ts#L122

**Comment by bpasero** [comment]
This looks like a regression from our Electron update where Squirrel is now taking longer to apply the update.

**Comment by chrisdias** [comment]
@joaomoreno I just ran into this with Insiders... is there no way to fix/patch in June?

**Comment by joaomoreno** [comment]
To keep the number of issues in our inbox on a manageable level, we're closing issues that have been on the backlog for a long time but haven't gained traction: We look at the number of votes the issue has received and the number of duplicates issues filed.

Thanks for your understanding. Happy coding!
